### 1.0 - Bibliotecas usadas.

In [1]:
import os
import numpy as np
from PIL import Image
import cv2

### 1.2
    Podemos ver os caminhos de todos os arquivos (que são imagens) que iremos usar.

In [2]:
print(os.listdir('./New_DB/yalefaces/train/'))

['subject01.glasses.gif', 'subject01.leftlight.gif', 'subject01.noglasses.gif', 'subject01.normal.gif', 'subject01.rightlight.gif', 'subject01.sad.gif', 'subject01.sleepy.gif', 'subject01.surprised.gif', 'subject01.wink.gif', 'subject02.glasses.gif', 'subject02.happy.gif', 'subject02.noglasses.gif', 'subject02.normal.gif', 'subject02.rightlight.gif', 'subject02.sad.gif', 'subject02.sleepy.gif', 'subject02.surprised.gif', 'subject02.wink.gif', 'subject03.centerlight.gif', 'subject03.happy.gif', 'subject03.noglasses.gif', 'subject03.normal.gif', 'subject03.rightlight.gif', 'subject03.sad.gif', 'subject03.sleepy.gif', 'subject03.surprised.gif', 'subject03.wink.gif', 'subject04.centerlight.gif', 'subject04.glasses.gif', 'subject04.happy.gif', 'subject04.noglasses.gif', 'subject04.normal.gif', 'subject04.rightlight.gif', 'subject04.sad.gif', 'subject04.sleepy.gif', 'subject04.wink.gif', 'subject05.centerlight.gif', 'subject05.glasses.gif', 'subject05.happy.gif', 'subject05.leftlight.gif', '

In [78]:
def get_image_data():
    paths = [os.path.join('./New_DB/yalefaces/train/', p) for p in os.listdir('./New_DB/yalefaces/train/')]  
    faces = []
    ids = []
    for paht in paths:
        #print(paht) # 1 - Para mostrar os caminhos.
        imagem = Image.open(paht).convert('L') # OBS: Convertendo a imagem para a escala de cinza.
        #print(type(imagem)) # 2 - Para mostrar os caminhos convertidos em imagem agora.
        imagem_np = np.array(imagem, 'uint8') #OBS: Convertendo para um npArray de inteiros.
        #print(type(imagem_np)) # 3 - Imagem convertidas em um array numpy - números
        #print(os.path.split(paht)[1]) # 4.1 - Primeira parte da extração do número id para 
        # associar o caminho a imagem
        #print(os.path.split(paht)[1].split('.')[0].split()) # 4.3 Mostrando como separar por nome.
        #print(os.path.split(paht)[1]) # 4.2 - Mesma coisa de cima, com apenas uma etapa a mais de extração.
        id = int(os.path.split(paht)[1].split('.')[0].replace('subject', ''))
        #print(id) # 5 - para mostrar a extrão do número id.
        ids.append(id) # 6 - Para assoscia o id a imagem.
        faces.append(imagem_np) # 7 - Imagem como forma de numpy array.
    return np.array(ids), faces

#### 2.0 - Pegando os dados das imagens preprocessadas.
    - 1 Agora para cada imagem, tem um id de identificação.
    - 2 Agora toda imagem já associada é um Numpy Array.

In [79]:
ids, faces = get_image_data()

### 2.1 - Agora vamos verificar os parâmetros até aqui.

In [80]:
print(len(ids))
print(len(faces))


135
135


#### 2.1.0
    - Temos 135 imagens de treino.

In [81]:
print(faces[0])
print(100*'*')
print('Tamanho da imagem - ',faces[0].shape)
a, b = faces[0].shape
print(100*'*')
print('Quantidade de Pixels de uma única imagem - ', a*b)

[[130 130 130 ... 255 255 255]
 [255 255 255 ... 255 255 255]
 [255 255 255 ... 255 255 255]
 ...
 [255 255 255 ... 252 255 255]
 [255 255 255 ... 255 255 255]
 [ 68  68  68 ...  68  68  68]]
****************************************************************************************************
Tamanho da imagem -  (243, 320)
****************************************************************************************************
Quantidade de Pixels de uma única imagem -  77760


### 2.1.1 - Com a função abaixo podemos ver a quantidade de pixels de cada imagem.
    - Obs: lembrando que só dar este valor a primeira imagem de 77760, porque está em escala de cinza.
    - Se não estivesse, seria x * y * z, onde z seria mais um número da escala de cores RGB.

In [82]:
def tamanho_das_imagens(face):
    for a, i in enumerate(face):
        #print(i)
        x, y = i.shape
        print(f'{a + 1} - ', x * y)

tamanho_das_imagens(faces)

1 -  77760
2 -  77760
3 -  77760
4 -  77760
5 -  77760
6 -  77760
7 -  77760
8 -  77760
9 -  77760
10 -  77760
11 -  77760
12 -  77760
13 -  77760
14 -  77760
15 -  77760
16 -  77760
17 -  77760
18 -  77760
19 -  77760
20 -  77760
21 -  77760
22 -  77760
23 -  77760
24 -  77760
25 -  77760
26 -  77760
27 -  77760
28 -  77760
29 -  77760
30 -  77760
31 -  77760
32 -  77760
33 -  77760
34 -  77760
35 -  77760
36 -  77760
37 -  77760
38 -  77760
39 -  77760
40 -  77760
41 -  77760
42 -  77760
43 -  77760
44 -  77760
45 -  77760
46 -  77760
47 -  77760
48 -  77760
49 -  77760
50 -  77760
51 -  77760
52 -  77760
53 -  77760
54 -  77760
55 -  77760
56 -  77760
57 -  77760
58 -  77760
59 -  77760
60 -  77760
61 -  77760
62 -  77760
63 -  77760
64 -  77760
65 -  77760
66 -  77760
67 -  77760
68 -  77760
69 -  77760
70 -  77760
71 -  77760
72 -  77760
73 -  77760
74 -  77760
75 -  77760
76 -  77760
77 -  77760
78 -  77760
79 -  77760
80 -  77760
81 -  77760
82 -  77760
83 -  77760
84 -  77760
8